### ISSUE TITLE
Elasticsearch should raise a mapper exception when reserved "mapping fields" are used inside "properties"

### ISSUE THREAD

**Issue by konradkonrad** [elastic/elasticsearch#10456]
Unfortunately it is possible to use ["top-level"/meta mapping properties](http://www.elastic.co/guide/en/elasticsearch/reference/1.x/mapping-fields.html), such as `_source`, `_all`, ... inside the `"properties": {}` portion of a mapping, without complaints. This allows for user errors that are easily missed.

Using the following mapping
```json
{
  "t1": {
    "_source": {"enabled": false}, 
    "properties": {
      "_all": {"enabled": false},
      "field1": {"type": "string"},
      "field2": {"type": "long"},
      "_underscore1": {"type": "string"},
      "_underscore2": {"enabled": true}
    }
  }
}
```
will lead to:

```json
{
  "mytest" : {
    "mappings" : {
      "t1" : {
        "_source" : {
          "enabled" : false
        },
        "properties" : {
          "_underscore1" : {
            "type" : "string"
          },
          "_underscore2" : {
            "type" : "object"
          },
          "field1" : {
            "type" : "string"
          },
          "field2" : {
            "type" : "long"
          }
        }
      }
    }
  }
}
```
Note, how the `_all`-"Field-Mapping" is just silently ignored.

Please see https://gist.github.com/konradkonrad/16220520eb9bacc1f6e6 for a complete reproduction.

**Comment by clintongormley** [comment]
Relates to #9059

**Comment by konradkonrad** [comment]
@clintongormley thx for labelling this! 

While I totally agree on the relation to #9059, I believe this one should/could be addressed ahead of time, since atm there exists a set of should-be-reserved words: [Mapping » Fields](http://www.elastic.co/guide/en/elasticsearch/reference/1.x/mapping-fields.html)*)

As the `_all` example seems to show, there is also some validation already in place. I cannot tell how many users use ambigous field names such as `_timestamp`, and I understand the BC concerns of #9059 -- however from *my* POV I'd rather have users become aware of their misconfigured mappings, then serve the BC needs of mappings using questionable ambigous names.

*) SN: btw. pretty unfortunate naming in the docs imho, what about one of these?
- *Mapping » DocType Configuration*
- *Mapping » DocType Settings*
- *Mapping » Mapping Configuration*
- *Mapping » Top-Level Fields*

**Comment by jpountz** [comment]
Fixed via #15243
